config/m4/knem.m4

```diff
@@ -0,0 +1,28 @@
+#
+# Check for KNEM support
+#
+knem_happy="no"
+AC_ARG_WITH([knem],
+           [AS_HELP_STRING([--with-knem=(DIR)], [Enable the use of KNEM (default is NO).])],
+           [], [with_knem=no])
+
+AS_IF([test "x$with_knem" != xno],
+      [AS_IF([test ! -d $with_knem], 
+             [
+              AC_MSG_NOTICE([KNEM path was not found, guessing ...])
+              with_knem=$(${PKG_CONFIG} --variable=prefix knem || find /opt/knem* -name knem_io.h |xargs dirname |sed -e s,/include,,g)
```

Comment:
i think we should add --with-knem to be the ON by default, if no path specified - it will look in /opt and will disqualify itself if not found.